JavaFiddle-Web/src/main/java/ru/javafiddle/web/services/ProjectService.java

```diff
@@ -30,16 +38,85 @@
     @EJB
     ProjectBean projectBean;
 
+    @EJB
+    FileBean fileBean;
+
+    @EJB
+    UserBean userBean;
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getUserProjectsList() {
+        String nickName = userBean.getCurUserNick();
+        List<String> projectHashes = userBean.getUserProjects(nickName);
+        String json = new Gson().toJson(projectHashes);
+        return Response.ok(json).build();
+    }
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("/{projectHash}")
+    public Response getProjectStructure(@PathParam("projectHash") String projectHash) throws Exception {
```

Comment:
This method is too heavy. Please split it into several ones (introduce auxiliary private methods).